packages/browser-agent-core/src/features/jserrors/aggregate/compute-stack-trace.test.js

```diff
@@ -0,0 +1,292 @@
+    }))
+    expect(result.frames).not.toContainEqual(expect.objectContaining({
+      func: 'nrWrapper'
+    }));
+  });
+
+  test("stack should still parse when column numbers are missing", () => {
+    const mockError = browserErrors.utils.constructError({
+      ...browserErrors.data.chromeEdge98Errors[0],
+      stack:
+        'Error: Blocked a frame with origin "http://bam-test-1.nr-local.net:3334" from accessing a cross-origin frame.\n    at errorTest (http://bam-test-1.nr-local.net:3334/tests/assets/instrumented.html?loader=spa:60)\n    at captureError (http://bam-test-1.nr-local.net:3334/tests/assets/instrumented.html?loader=spa:17)\n    at onload (http://bam-test-1.nr-local.net:3334/tests/assets/instrumented.html?loader=spa:57)',
+    });
+
+    const result = computeStackTrace(mockError);
+
+    expect(result).toEqual(expect.objectContaining({
+      mode: 'stack',
+      name: mockError.name,
+      message: mockError.message,
+      stackString: mockError.stack
+    }))
+    expect(result.frames.length).toEqual(3);
+    expect(result.frames).toContainEqual(expect.objectContaining({
+      func: 'errorTest',
+      column: null
+    }));
+    expect(result.frames).toContainEqual(expect.objectContaining({
+      func: 'captureError',
+      column: null
+    }));
+    expect(result.frames).toContainEqual(expect.objectContaining({
+      func: 'onload',
+      column: null
+    }));
+  });
+
+  test("parser can handle chrome eval stack", () => {
+    const mockError = browserErrors.utils.constructError({
+      ...browserErrors.data.chromeEdge98Errors[0],
+      stack:
+        "    at foobar (eval at foobar (http://bam-test-1.nr-local.net:3334/tests/assets/instrumented.html))",
+    });
+
+    const result = computeStackTrace(mockError);
+
+    expect(result).toEqual(expect.objectContaining({
+      mode: 'stack',
+      name: mockError.name,
+      message: mockError.message,
+      stackString: mockError.stack
+    }))
+    expect(result.frames.length).toEqual(1);
+    expect(result.frames).toContainEqual(expect.objectContaining({
+      func: 'evaluated code'
+    }));
+  });
+
+  test("parser can handle ie eval stack", () => {
+    const mockError = browserErrors.utils.constructError({
+      ...browserErrors.data.ie11Errors[0],
+      stack:
+        "    at Function code (Function code:23:23)",
+    });
+
+    const result = computeStackTrace(mockError);
+
+    expect(result).toEqual(expect.objectContaining({
+      mode: 'stack',
+      name: mockError.name,
+      message: mockError.message,
+      stackString: mockError.stack
+    }))
+    expect(result.frames.length).toEqual(1);
+    expect(result.frames).toContainEqual(expect.objectContaining({
+      func: 'evaluated code'
+    }));
+  });
+
+  test("parser can handle stack with anonymous function", () => {
+    const mockError = browserErrors.utils.constructError({
+      ...browserErrors.data.chromeEdge98Errors[0],
```

Comment:
The way fixture data is used and is intended to be used as time goes on in this test is a little confusing. Are these values supposed to exist even as our testing window slides (at some point (soon) we won't support testing chrome 98, ie11, etc)

When do we update this list of fixture data and how?

Is there a way we can represent the fixture data in a more static way that stays evergreen?

Reply:
The data directory contains a README with details on how the test data was generated. We could take a subset of the data and remove the rest